ccp/control.sls

```diff
@@ -31,6 +31,32 @@ ccp_repos_dir:
   - require:
     - user: ccp_user
 
+ccp_ssh_dir:
+  file.directory:
+  - names:
+    - {{ control.dir.base }}/.ssh/
+  - mode: 700
+  - makedirs: true
+  - user: ccp
+  - require:
+    - user: ccp_user
+
+ccp_ssh_key:
+  cmd.run:
+  - name: ssh-keygen -q -N '' -f /srv/ccp/.ssh/id_rsa
```

Comment:
I think you need to know private/public key in advance before deployment to be able to setup github user or deploy key.
Common practice is to generate ssh key and put it into metadata model. openssh formula then takes care of putting it into right location for the right user + the same for known hosts hashing.
So use openssh:client instead of replicating job of another formula. https://github.com/salt-formulas/salt-formula-openssh